src/goto-programs/html_goto_trace.cpp

```diff
@@ -0,0 +1,278 @@
+/*******************************************************************\
+
+Module: HTML Traces of GOTO Programs
+
+Author: elizabeth.polgreen@cs.ox.ac.uk
+
+  Date: October 2017
+
+\*******************************************************************/
+
+/// \file
+/// HTML traces of GOTO Programs
+
+#include "goto_trace.h"
+
+#include <cassert>
+#include <ostream>
+
+#include <util/arith_tools.h>
+#include <util/config.h>
+#include <util/symbol.h>
+
+#include <ansi-c/printf_formatter.h>
+#include <langapi/language_util.h>
+
+
+void output_html_header(std::ostream &out)
+{
+  out << "<!DOCTYPE html> \n" << "<html> \n" << "<head> \n" << "<style> \n"
+      << "button.function_call{ background-color: #eee; color: Black; cursor: pointer; \n"  // NOLINT(*)
+      << "padding: 18px; width: 100%; border: double; text-align: left; outline: none; \n" // NOLINT(*)
+      << "font-size: 15px; transition: 0.4s; } \n\n"
+      << "button.function_call.active, button.fuction_call:hover { background-color: LightGray; }\n"  // NOLINT(*)
+
+      << "div.panel1 { padding: 0 18px; display: none; background-color: AliceBlue; border: 1px solid gray } \n"  // NOLINT(*)
+      << "div.panel2 { padding: 0 18px; display: none; background-color: CornSilk; border: 1px solid gray} \n"  // NOLINT(*)
+      << "div.property_panel { padding: 0 18px; display: block; background-color: LightGray; border: 1px red} \n"  // NOLINT(*)
+
```

Comment:
If we're doing style we should probably do so everywhere, and turn e.g. `<strong><span style=\"color: Red;\">` below into `<span class=\"violated_property\">`, rather than intermixing out-of-line and inline style.